### PR TITLE
fix(github): Disable the integration on token creation error

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

**Review by dcramer — CHANGES_REQUESTED** [review]
can we quickly whip up an email for this situation? similar to how we email them when we fail to fetch commits in the releases

**Comment by EvanPurkhiser** [comment]
@macqueen were you still planning to work on something like this in the near term? I can add a quick email for this if you think it's something that's going to happen in less-near-term.

File: src/sentry/integrations/github/client.py [hunk 1]
```diff
@@ -112,7 +114,18 @@ def get_token(self):
...
-            res = self.create_token()
+            try:
+                res = self.create_token()
+            except ApiError:
+                # The token could not be refreshed. This likely indicates the
+                # installation was removed but failed to be disabled through
+                # the uninstall webhook. disable the integration now.
+                #
+                # TODO(epurkhiser): When we have some kind of logging mechanism
+                # for the user to inspect for integrations, log something here.
```
Comment (markstory) [inline src/sentry/integrations/github/client.py]: This sounds like a great improvement for integrations.

Comment (lauryndbrown) [reply src/sentry/integrations/github/client.py]: @EvanPurkhiser would audit log not be useful here?

**Pull request closed without merging**